### PR TITLE
Remove the `hasNans` config and update the doc

### LINKED ISSUE #6487: [FEA] Remove the `hasNans` config from the test code.
**Is your feature request related to a problem? Please describe.**
After finishing #5320, we should remove all `hasNans` config in the tests.

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**
Add any other context, code examples, or references to existing implementations about the feature request here.

### LINKED ISSUE #6488: [FEA] Remove the `hasNans` configuration and update the docs
**Is your feature request related to a problem? Please describe.**
After finishing #5320, we should remove the `hasNans` config.
We should also update documentation to remove the content related to `hasNans`

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**
Add any other context, code examples, or references to existing implementations about the feature request here.

### PR DESCRIPTION
**Pull request by HaoYang670**
Signed-off-by: remzi <13716567376yh@gmail.com>
Close #6488.
Close #6487

### CODE REVIEW

**Comment by HaoYang670** [comment]
build

**Comment by HaoYang670** [comment]
build

**Comment by jlowe** [comment]
There are also references to the `hasNans` config in some of the notebooks under the docs/demo directory:
```
$ git grep hasNans docs/demo
docs/demo/AWS-EMR/Mortgage-ETL-GPU-EMR.ipynb:    "    \"conf\": {\"spark.sql.adaptive.enabled\": \"false\", \"spark.dynamicAllocation.enabled\": \"false\", \"spark.executor.instances\":2, \"spark.executor.cores\":2, \"spark.rapids.sql.explain\":\"ALL\", \"spark.task.cpus\":\"1\", \"spark.rapids.sql.concurrentGpuTasks\":\"2\", \"spark.rapids.memory.pinnedPool.size\":\"2G\", \"spark.executor.memoryOverhead\":\"2G\", \"spark.executor.extraJavaOptions\":\"-Dai.rapids.cudf.prefer-pinned=true\", \"spark.sql.files.maxPartitionBytes\":\"512m\", \"spark.executor.resource.gpu.amount\":\"1\", \"spark.task.resource.gpu.amount\":\"0.5\", \"spark.plugins\":\"com.nvidia.spark.SQLPlugin\", \"spark.rapids.sql.hasNans\":\"false\", \"spark.rapids.sql.batchSizeBytes\":\"512M\", \"spark.rapids.sql.reader.batchSizeBytes\":\"768M\", \"spark.sql.cache.serializer\" : \"com.nvidia.spark.ParquetCachedBatchSerializer\"}\n",
docs/demo/Databricks/Mortgage-ETL-db.ipynb:    "spark.conf.set(\"spark.rapids.sql.hasNans\", \"false\")\n",
docs/demo/GCP/Mortgage-ETL.ipynb:    "conf.set(\"spark.rapids.sql.hasNans\", \"false\")\n",
docs/demo/GCP/Mortgage-ETL.ipynb:    "spark.conf.set(\"spark.rapids.sql.hasNans\", \"false\")\n",
docs/demo/GCP/mortgage-xgboost4j-gpu-scala.ipynb:    "sparkSession.conf.set(\"spark.rapids.sql.hasNans\", false)\n",
```

File: docs/demo/Databricks/Mortgage-ETL-db.ipynb [hunk 1]
```diff
@@ -17,7 +17,7 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 0,
+   "execution_count": null,
```
Comment (HaoYang670) [inline docs/demo/Databricks/Mortgage-ETL-db.ipynb:20]: I don't know why this line is modified. What I did is just removing the line containing `hasNan`

**Comment by HaoYang670** [comment]
I also removed the `hasNan` config from the spark-2-plugin.

**Comment by HaoYang670** [comment]
build

File: docs/compatibility.md [hunk 2]
```diff
@@ -69,18 +69,11 @@ worked with plain Spark. Starting from 22.06 this is behavior is enabled by defa
 the config
 [`spark.rapids.sql.variableFloatAgg.enabled`](configs.md#sql.variableFloatAgg.enabled).
 
-Additionally, some aggregations on floating point columns that contain `NaN` can produce results
-different from Spark in versions prior to Spark 3.1.0.  If it is known with certainty that the
-floating point columns do not contain `NaN`, set
-[`spark.rapids.sql.hasNans`](configs.md#sql.hasNans) to `false` to run GPU enabled aggregations on
-them.
-
 In the case of a distinct count on `NaN` values, prior to Spark 3.1.0, the issue only shows up if
```
[the remark this answers is not in the document]
Comment (HaoYang670) [reply docs/compatibility.md]: Removed.

**Comment by HaoYang670** [comment]
build

**Review by jlowe — APPROVED** [review]

**Pull request merged**